### PR TITLE
openjdk 15.0.2

### PR DESCRIPTION
**Pull request by el7cosmos**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

File: Formula/openjdk.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ class Openjdk < Formula
     sha256 "29df31b5eefb5a6c016f50b2518ca29e8e61e3cfc676ed403214e1f13a78efd5"
     version "15.0.1"
```
Comment (carlocab) [inline Formula/openjdk.rb:9]:
```suggestion
    version "15.0.2"
```
This version string also needs to be changed, or else `test-bot` won't be able to find the ARM bottle

File: Formula/openjdk.rb [hunk 2]
```diff
@@ -8,8 +8,8 @@ class Openjdk < Formula
...
   else
-    url "https://hg.openjdk.java.net/jdk-updates/jdk15u/archive/jdk-15.0.1-ga.tar.bz2"
-    sha256 "9c5be662f5b166b5c82c27de29b71f867cff3ff4570f4c8fa646490c4529135a"
+    url "https://hg.openjdk.java.net/jdk-updates/jdk15u/archive/jdk-15.0.2-ga.tar.bz2"
+    sha256 "1ab1de3f4c8471e9eaea1479159dcc34266ea17f7fec30f7ed4ac204bafc11cd"
```
Comment (carlocab) [inline Formula/openjdk.rb:12]:
```suggestion
    sha256 "d07b45b5b319e7034e8ebc41cd78c496e6ee8b1f6e08310dee303beaee8b4a3a"
```

**Comment by carlocab** [comment]
Patches look like they're no longer necessary:
https://bugs.openjdk.java.net/browse/JDK-8253375
https://bugs.openjdk.java.net/browse/JDK-8253791

**Comment by carlocab** [comment]
All three Intel CI nodes:
```
Error: An exception occurred within a child process:
  RuntimeError: cannot find build number in .hg_archival.txt
```

**Comment by carlocab** [comment]
For openjdk 15.0.1, this is the content of `.hg_archival.txt`:
```
repo: fd16c54261b32be1aaedd863b7e856801b7f8543
node: d9b7c07593100c3cae95238999f972ceb0ff067c
branch: default
tag: jdk-15.0.1+9
tag: jdk-15.0.1-ga
```
This gives the build number `9`. If we had used the same [scheme](https://github.com/Homebrew/homebrew-core/blob/f9b4c5b0b556f4f2845735c101edd597e2abcf1a/Formula/openjdk.rb#L98-L105) as ARM to parse the build number from `.hgtags`, we'd have gotten `8`:
```
❯ tail -10 .hgtags
e3f940bd3c8fcdf4ca704c6eb1ac745d155859d5 jdk-15+36
93813843680bbe1b7efbca56c03fd137f20a2c31 jdk-15.0.1+0
286ddf41c3f697bcf684042544fa4736ba1b85f3 jdk-15.0.1+1
08e8fa30d8456913dd0f698f9c61e4adac22805a jdk-15.0.1+2
99d9a174da61ca1e359638170b47a7ddc9cc77d5 jdk-15.0.1+3
72cddf95705e758aea928e025694e445af844d1f jdk-15.0.1+4
ac7b0cbc8dac94ff225021d2786efbad2326b037 jdk-15.0.1+5
a90f4378bf70eb17e8a8322aa0b94a4c6c923932 jdk-15.0.1+6
23fd169be90a803fef99776afefa3072189980be jdk-15.0.1+7
8dd4d8eaf0de32a79d1f48c3421e5666d75beff9 jdk-15.0.1+8
```

**Comment by SMillerDev** [comment]
Why do we need to have the build number?

If it's useful I guess we could move to the arm way (since that's where they seem to be going)

**Comment by carlocab** [comment]
>Why do we need to have the build number?

No idea.

>we could move to the arm way

That would've given a build number of 8 instead of 9 for 15.0.1 though: https://github.com/Homebrew/homebrew-core/pull/69460#issuecomment-764445006 Not sure if that's important at all.

**Comment by jonchang** [comment]
It "should" be +8. Usually they have two tags pointing to a single commit like so:

```
e3f940bd3c8fcdf4ca704c6eb1ac745d155859d5 jdk-15+36
e3f940bd3c8fcdf4ca704c6eb1ac745d155859d5 jdk-15-ga
```

for some reason upstream has not done it this time. We could potentially wait for adoptopenjdk to release 15.0.2 to see what build number they pick.

**Comment by jonchang** [comment]
> Why do we need to have the build number?
> 
> If it's useful I guess we could move to the arm way (since that's where they seem to be going)

Let's just go with this since it's easier. I think build numbers are really only used when comparing among java versions and I think it's a legacy of openjdk 8-era java versioning (like jdk8u282)

**Comment by carlocab** [comment]
>Let's just go with this since it's easier.

Done. This is still slightly different from ARM, which uses only the major version, though. However, I assume that's because `.hgtags` in the ARM tarball only contains lines with the major version (unlike the 15.0.2 tarball).

**Comment by carlocab** [comment]
Same error; still can't find the build number. (I forgot to change the error message to `.hgtags`.) Not sure why; these are the relevant lines in `.hgtags`:
```
❯ tail -8 .hgtags
1613004c47e9dc867a2c2c43d716533b1aaedc5f jdk-15.0.2+0
cc4fdb537bc14734064a9a8eadb091fd1c12b36e jdk-15.0.2+1
d24e907486b3f90691980b0dde01efca5840abc6 jdk-15.0.2+2
dbb11e11955ad1240ba775ab0007a14547e14ce6 jdk-15.0.2+3
4c4a2eb7b19ecb31620e6bb120e40f8a5fd1737a jdk-15.0.2+4
e431a9461b1356c4b763443e5333b3f4a8695eaf jdk-15.0.2+5
d5977ee56509ceaa3d3c8e1aebbca76651358da4 jdk-15.0.2+6
38912b2a5bcb396c75f8707e300773c874327451 jdk-15.0.2+7
```

**Comment by carlocab** [comment]
Re the build number: https://github.com/Homebrew/homebrew-core/pull/68946#discussion_r557588659

It seems we should add one to it on Intel? (I'm not even sure how important this is.)

**Comment by carlocab** [comment]
ARM CI broken by https://github.com/Homebrew/brew/pull/10375.
```
Error: Invalid usage: not (yet) working on Apple Silicon!
```

**Comment by carlocab** [comment]
Big Sur

```
Error: 12 failed steps!
brew test --retry --verbose byteman
brew install --build-from-source emscripten
brew test --retry --verbose fits
brew test --retry --verbose jasmin
brew test --retry --verbose jetty
brew install --build-from-source languagetool
brew install --build-from-source man-db
brew install --build-from-source mesos
brew test --retry --verbose nexus
rm -rf /Users/brew/Library/Caches/Homebrew
brew install --build-from-source planck
brew install --build-from-source simutrans
```

Catalina

```
Error: 8 failed steps!
brew test --retry --verbose byteman
brew test --retry --verbose fits
brew test --retry --verbose jasmin
brew test --retry --verbose jetty
brew install --build-from-source man-db
brew install --build-from-source mesos
brew install --build-from-source planck
brew install --build-from-source simutrans
```

Mojave

```
Error: 5 failed steps!
brew test --retry --verbose byteman
brew test --retry --verbose fits
brew test --retry --verbose jasmin
brew test --retry --verbose jetty
brew install --build-from-source planck
```

**Comment by jonchang** [comment]
Looks like adoptopenjdk is using 15.0.2+7 🤷

**Comment by jonchang** [comment]
mesos and man-db are now fixed, so hopefully this CI run works.

**Comment by jonchang** [comment]
Mojave is 🍏 🙏

**Comment by carlocab** [comment]
Big Sur:

```
Error: 4 failed steps!
rm -rf /Users/brew/Library/Caches/Homebrew
brew install --build-from-source emscripten
rm -rf /Users/brew/Library/Caches/Homebrew
brew install --build-from-source varnish
```

Catalina:

```
Error: 2 failed steps!
brew install --build-from-source root
brew install --build-from-source varnish
```

`emscripten`:

```
cp: /private/tmp/emscripten--llvm-20210302-46554-1i290gk/.git/packed-refs: No space left on device
```

`varnish`: #72404

`root`:

```
==> brew install --build-from-source root
==> FAILED
==> Downloading https://root.cern.ch/download/root_v6.22.06.source.tar.gz
curl: (56) LibreSSL SSL_read: error:06FFF064:digital envelope routines:CRYPTO_internal:bad decrypt, errno 0
Error: Failed to download resource "root"
Download failed: https://root.cern.ch/download/root_v6.22.06.source.tar.gz
```

**Comment by carlocab** [comment]
ARM runner is still going. It seems to be weird to have to wait for it, given that this PR should have no impact on the ARM bottle other than renaming it...

**Comment by jonchang** [comment]
I'm just going to publish these bottles. OpenJDK 16 is coming out in a few weeks so we need to direct our energy there :shrug:

**Comment by jonchang** [comment]
I logged into the ARM runner and determined that an automatic update (of the runner) seemed to have killed the runner instance, which is why the tests were stalled.

**Pull request closed without merging**